shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -406,7 +406,6 @@ - (void)loadAOTData:(NSString*)assetsDir {
 - (void)setViewController:(FlutterViewController*)controller {
   if (_viewController != controller) {
     _viewController = controller;
-    [_renderer setFlutterView:controller.flutterView];
```

Comment:
Where is the view registered with the view provider?

Reply:
Any other places. The view provider is just a protocol/facade over other classes. It will have other methods to register views, but not as a view provider.
In the real application the view provider is `macos/FlutterEngine`, which is the source-of-truth of the set of views.